Gems/Atom/Tools/MaterialEditor/Code/Source/Viewport/MaterialViewportWidget.cpp

```diff
@@ -69,9 +69,15 @@ namespace MaterialEditor
 
         // The viewport context created by AtomToolsFramework::RenderViewportWidget has no name.
         // Systems like frame capturing and post FX expect there to be a context with DefaultViewportContextName
-        auto viewportContextManager = AZ::Interface<AZ::RPI::ViewportContextRequestsInterface>::Get();
-        const AZ::Name defaultContextName = viewportContextManager->GetDefaultViewportContextName();
-        viewportContextManager->RenameViewportContext(GetViewportContext(), defaultContextName);
+        // ------------------------------
+        // TODO: When MaterialViewportWidget is opened inside the Editor as a dll it already exists
+        //       a viewport context with the default viewport context name. It asserts if it cannot
+        //       be renamed or for now keeping the original viewport context name. This has to be
+        //       handled properly in the future.
+        // ------------------------------
+        //auto viewportContextManager = AZ::Interface<AZ::RPI::ViewportContextRequestsInterface>::Get();
+        //const AZ::Name defaultContextName = viewportContextManager->GetDefaultViewportContextName();
+        //viewportContextManager->RenameViewportContext(GetViewportContext(), defaultContextName);
```

Comment:
@gadams3 for the standalone application it's ok to rename the viewport context to the `defaultContextName`. 
But when this runs within the editor, there is already another viewport context called `defaultContextName` and therefore this one fails and crashes.

What are the implications of this viewport context not being called `defaultContextName`?

What would you recommend on this issue?

Reply:
it was renamed to the default because it was not going through the bootstrap component that normally does this and some system required it. I don't remember all of the details. Should be in the comments. Does not matter to me what the name is as long as it is unique in case there is more than one, probably generated or appended with a UUID.

Reply:
>Does not matter to me what the name is as long as it is unique in case there is more than one, probably generated or appended with a UUID

By taking a look at the code the name is already unique per viewport context.

`MaterialViewportWidget` inherits from `RenderViewportWidget`, which creates the viewport context in `InitializeViewportContext` by calling `ViewportContextManager::CreateViewportContext` with an empty name and `AzFramework::InvalidViewportId`. 

`ViewportContextManager::CreateViewportContext` then will provide an unique id incrementing `m_currentViewportId` and an unique name from it `Name(AZStd::string::format("ViewportContext%i", id))`.